vividus-plugin-json/src/main/java/org/vividus/json/steps/JsonSteps.java

```diff
@@ -363,16 +368,24 @@ private void executeStepsForFoundJsonElements(String json, String jsonPath, Comp
     public void assertValueByJsonPath(String json, String jsonPath, String comparisonRule, Object expectedData)
     {
         getDataByJsonPathSafely(json, jsonPath, true).ifPresent(jsonByPath -> {
+            List<String> equalAndNotEqualComparisonRules = Arrays.asList("IS_EQUAL_TO", "IS_NOT_EQUAL_TO");
             if (jsonByPath.isEmpty())
             {
-                validateEqualsComparison(comparisonRule, null, expectedData);
-                assertThat(jsonPath, null, equalTo(expectedData));
+                validateComparisonRuleAndNullValue(jsonPath, comparisonRule, equalAndNotEqualComparisonRules,
+                        null, expectedData);
                 return;
             }
             Object jsonValue = jsonByPath.get();
+            if (expectedData == null)
+            {
+                validateComparisonRuleAndNullValue(jsonPath, comparisonRule, equalAndNotEqualComparisonRules,
+                        jsonValue, null);
+                return;
+            }
             if (jsonValue instanceof Boolean)
             {
-                validateEqualsComparison(comparisonRule, jsonValue, expectedData);
+                validateComparisonRule(comparisonRule, equalAndNotEqualComparisonRules.subList(0, 1),
```

Comment:
😮 sublist to get a single element (which is actually constant)?

Reply:
fixed